gateway/shard/connect.ts

```diff
@@ -8,8 +8,7 @@ export async function connect(shard: Shard): Promise<void> {
   }
   shard.events.connecting?.(shard);
 
-  // Explicitly setting the encoding to json, since we do not support ETF.
-  const socket = new WebSocket(`${shard.gatewayConfig.url}/?v=${shard.gatewayConfig.version}&encoding=json`);
+  const socket = new WebSocket(shard.resumeGatewayUrl);
```

Comment:
This should be like a ternary that checks if shard.state is Resuming and only then it uses this resume gateway url otherwise it should use the old one, also either way we need to keep the v=${version}&encoding=json